wonsik/app.js

```diff
@@ -40,14 +40,169 @@ app.post("/users", async (req, res) => {
       email,
       profile_image,
       password
-    )VALUES (?, ?, ?, ?);
+    )VALUES(?, ?, ?, ?);
     `,
     [name, email, profileImage, password]
   );
 
   res.status(201).json({ message: "userCreated" });
 });
 
+//Create a post
+app.post("/posts", async (req, res) => {
+  const { title, content, imageUrl, userId } = req.body;
+
+  await appDataSource.query(
+    `INSERT INTO posts(
+      title,
+      content,
+      image_url,
+      user_id
+    )VALUES (?, ?, ?, ?);
+    `,
+    [title, content, imageUrl, userId]
+  );
+
+  res.status(201).json({ message: "postCreated" });
+});
+
+// assignment 4
+app.get("/postlist", async (req, res) => {
+  const result = await appDataSource.query(
+    `
+    SELECT
+      u.id            as userId,
+      u.profile_image as userProfileImage,
+      p.id            as postId,
+      p.image_url     as postImageUrl,
+      p.title         as postTitle,
+      p.content       as postContent
+    FROM posts as p
+    JOIN users as u ON u.id = p.user_id
+  `
+  );
+
+  return res.status(200).json({ data: result });
+});
+
+// 5번
+app.get("/users/:userId/posts", async (req, res) => {
+  const { userId } = req.params;
+
+  const result = await appDataSource.query(
+    `
+  SELECT
+  u.id as userId,
+  u.profile_image as userProfileImage,
+  JSON_ARRAYAGG(
+    JSON_OBJECT(
+      "postingId", p.id,
+      "postingImageUrl", p.image_url,
+      "postingTitle", p.title,
+      "postingContent", p.content
+    )
+  ) as postings
+  FROM users as u
+  JOIN posts as p ON p.user_id = u.id
+  WHERE u.id = ?
+  GROUP BY u.id
+  `,
+    [userId]
+  );
+
+  return res.status(200).json({ data: result });
+});
+
+// 6번
+app.patch("/posts/:postId", async (req, res) => {
+  const { postId } = req.params;
+  const { content } = req.body;
+
+  await appDataSource.query(
+    `
+    UPDATE posts
+    SET content = ?
+    WHERE id = ?
+    `,
+    [content, postId]
+  );
+
+  const result = await appDataSource.query(
+    `
+    SELECT
+     u.id      as userId,
+     u.name    as userName,
+     p.id      as postId,
+     p.title   as postTitle,
+     p.content as postContent
+    FROM posts p
+    JOIN users u ON u.id = p.user_id
+    WHERE p.id = ?
+    `,
+    [postId]
+  );
+
+  return res.status(201).json({ data: result });
+});
```

Comment:
1. 해당 api는 update 후 update 된 post를 반환하는 것으로 보입니다. 1개의 데이터를 보낼 때는 배열에 담기보다는 객체로 보내는 것이 더 좋습니다. 따라서 예시 코드를 보면 result라는 변수를 배열로 감쌌는데, 이렇게하면 result라는 변수에 결과로 올 배열의 첫번째 값이 result 변수에 담깁니다. 그러면 배열에 담긴 객체가 아닌 단순 객체로 보낼 수 있습니다.
2. status code 201은 데이터가 생성 될 때 사용하므로 200으로 수정해 주면 좋을 것 같습니다.
```suggestion
  const [result] = await appDataSource.query(
    `
    SELECT
     u.id      as userId,
     u.name    as userName,
     p.id      as postId,
     p.title   as postTitle,
     p.content as postContent
    FROM posts p
    JOIN users u ON u.id = p.user_id
    WHERE p.id = ?
    `,
    [postId]
  );

  return res.status(200).json({ data: result });
});
```